sigar_windows.go

```diff
@@ -295,28 +308,304 @@ func (self *DiskList) Get() error {
+	WindowsRunStateGrowing
+)
+
+func convertWindowsRunState(state WindowsRunState) RunState {
+	// This mapping may not be exact, see "ExecutionState" at
+	// https://msdn.microsoft.com/en-us/library/aa387976(v=vs.85).aspx
+	switch WindowsRunState(state) {
+	case WindowsRunStateReady:
+	case WindowsRunStateRunning:
+		return RunStateRun
+	case WindowsRunStateBlocked:
+	case WindowsRunStateSuspendedBlocked:
+		return RunStateIdle
+	case WindowsRunStateTerminated:
+		return RunStateZombie
+	case WindowsRunStateStopped:
+		return RunStateStop
+	}
+	return RunStateUnknown
+}
+
+func (self *ProcessList) Get() error {
+	var procs []Win32_Process
+	query := wmi.CreateQuery(&procs, "")
+	err := wmiClient.Query(query, &procs)
+	if err != nil {
+		return err
+	}
+
+	processes := make([]Process, 0, len(procs))
+	for _, proc := range procs {
+		var process Process
+
+		// ProcState
+		process.ProcState.Name = proc.Name
+		process.ProcState.Pid = int(proc.ProcessId)
+		process.ProcState.Ppid = int(proc.ParentProcessId)
+		process.ProcState.Priority = int(proc.Priority)
+		process.ProcState.State = convertWindowsRunState(WindowsRunState(proc.ExecutionState))
+
+		// ProcIo
+		process.ProcIo.ReadBytes = proc.ReadTransferCount
+		process.ProcIo.ReadOps = proc.ReadOperationCount
+		process.ProcIo.WriteBytes = proc.WriteTransferCount
+		process.ProcIo.WriteOps = proc.WriteOperationCount
+
+		// ProcMem
+		process.ProcMem.Size = proc.VirtualSize
+		process.ProcMem.Resident = proc.WorkingSetSize
+		process.ProcMem.PageFaults = uint64(proc.PageFaults)
+
+		// ProcTime
+		if proc.UserModeTime != 0 {
+			// Time in user mode, in 100 nanosecond units. Divide by 10000 to get milliseconds
+			process.ProcTime.User = proc.UserModeTime / 10000
+		}
+		if proc.KernelModeTime != 0 {
+			// Time in kernel mode, in 100 nanosecond units. Divide by 10000 to get milliseconds
+			process.ProcTime.Sys = proc.KernelModeTime / 10000
+		}
+		process.ProcTime.Total = process.ProcTime.User + process.ProcTime.Sys
+		// Convert proc.CreationDate to millis
+		process.ProcTime.StartTime = uint64(proc.CreationDate.UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond)))
+
+		// ProcArgs
+		process.ProcArgs.List = strings.Split(proc.CommandLine, " ")
+
+		// ProcExe - Cwd, Root not implemented
+		process.ProcExe.Name = proc.ExecutablePath
+
+		processes = append(processes, process)
+	}
+	self.List = processes
+	return nil
+}
+
 func (self *ProcList) Get() error {
-	return notImplemented()
+	var procs []Win32_Process
+	query := wmi.CreateQuery(&procs, "")
+	err := wmiClient.Query(query, &procs)
+	if err != nil {
+		return err
+	}
+
+	pids := make([]int, 0, len(procs))
+	for _, proc := range procs {
+		pids = append(pids, int(proc.ProcessId))
+	}
+	self.List = pids
+	return nil
+}
+
+// Helper to consistently handle WMI results
+func validateSingleWmiResult(pid int, results interface{}) error {
+	switch res := results.(type) {
+	case []Win32_Process:
+		if len(res) == 0 {
```

Comment:
At this point, does `res` equal the type of `results`, or `results` cast to a `[]Win32_Process`?  I'm guessing the latter, or else this probably wouldn't compile :)